buildSrc/src/main/kotlin/com/squareup/anvil/benchmark/CreateBenchmarkProjectTask.kt

```diff
@@ -0,0 +1,392 @@
+package com.squareup.anvil.benchmark
+
+import com.squareup.anvil.benchmark.Module.AppModule
+import com.squareup.anvil.benchmark.Module.LibraryModule
+import com.squareup.kotlinpoet.AnnotationSpec
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier.ABSTRACT
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asClassName
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+private const val MODULE_COUNT = 1000
+
+/**
+ * Generates 1000 library modules and 1 application module to benchmark Anvil. The first 10
+ * generated library modules are special, because they define the interfaces that are bound by
+ * other implementation classes in the other library modules. These 10 library modules also
+ * contribute one subcomponent each.
+ *
+ * All library modules contribute a binding and Dagger module. The contributed binding replaces
+ * the contributed binding from the module with the index - 10 to avoid duplicate bindings. The
+ * Dagger module has one provider method for another interface. We replace all Dagger modules
+ * except for the last 10 in the application module to avoid duplicate bindings.
+ *
+ * The app module contains 10 Dagger components with the same scope and runs the Dagger
+ * annotation processor with KAPT. Each of the Dagger components merges the 10 contributed
+ * subcomponents, so in total we have 100 subcomponents. For each of them + the 10 components
+ * we run Anvil's slowest operations of merging all contributions.
```

Comment:
I tried to describe what happens. In practice, it looks like this. For the first 10 modules we generate:
```kotlin
public interface BoundInterface0

@ContributesTo(scope = Unit::class)
public interface ContributedInterfaceComponent0 {
  public fun boundInterface0(): BoundInterface0
}

public interface InjectedInterface0

@ContributesSubcomponent(
  scope = Any::class,
  parentScope = Unit::class
)
public interface ContributedSubcomponent0

@ContributesTo(scope = Any::class)
public interface ContributedInterfaceSubcomponent0 {
  public fun injectedInterface0(): InjectedInterface0
}

@ContributesBinding(scope = Unit::class)
public object BoundInterfaceImpl0 : BoundInterface0

public class InjectedInterfaceImpl0 @Inject constructor(
  @Suppress("UNUSED_PARAMETER")
  boundInterface: BoundInterface0
) : InjectedInterface0

@Module
@ContributesTo(scope = Any::class)
public object DaggerModule0 {
  @Provides
  public fun provideInjectedInterface(injectedInterfaceImpl: InjectedInterfaceImpl0):
      InjectedInterface0 = injectedInterfaceImpl
}
```
The other modules look like this:
```
@ContributesBinding(
  scope = Unit::class,
  replaces = [BoundInterfaceImpl0::class, BoundInterfaceImpl10::class]
)
public object BoundInterfaceImpl20 : BoundInterface0

public class InjectedInterfaceImpl20 @Inject constructor(
  @Suppress("UNUSED_PARAMETER")
  boundInterface: BoundInterface0
) : InjectedInterface0

@Module
@ContributesTo(scope = Any::class)
public object DaggerModule20 {
  @Provides
  public fun provideInjectedInterface(injectedInterfaceImpl: InjectedInterfaceImpl20):
      InjectedInterface0 = injectedInterfaceImpl
}
```
And finally the app module contains:
```kotlin
@MergeComponent(scope = Unit::class)
public interface Component1

@MergeComponent(scope = Unit::class)
public interface Component2

@MergeComponent(scope = Unit::class)
public interface Component3

@MergeComponent(scope = Unit::class)
public interface Component4

@MergeComponent(scope = Unit::class)
public interface Component5

@MergeComponent(scope = Unit::class)
public interface Component6

@MergeComponent(scope = Unit::class)
public interface Component7

@MergeComponent(scope = Unit::class)
public interface Component8

@MergeComponent(scope = Unit::class)
public interface Component9

@MergeComponent(scope = Unit::class)
public interface Component10

@Module
@ContributesTo(
  scope = Any::class,
  replaces = [DaggerModule0::class, DaggerModule1::class, DaggerModule2::class,
      DaggerModule3::class, DaggerModule4::class, DaggerModule5::class, DaggerModule6::class,
      DaggerModule7::class, DaggerModule8::class, DaggerModule9::class, DaggerModule10::class,
      DaggerModule11::class, DaggerModule12::class, DaggerModule13::class, DaggerModule14::class,
      DaggerModule15::class, DaggerModule16::class, DaggerModule17::class, DaggerModule18::class,
      DaggerModule19::class, DaggerModule20::class, DaggerModule21::class, DaggerModule22::class,
      DaggerModule23::class, DaggerModule24::class, DaggerModule25::class, DaggerModule26::class,
      DaggerModule27::class, DaggerModule28::class, DaggerModule29::class, DaggerModule30::class,
      DaggerModule31::class, DaggerModule32::class, DaggerModule33::class, DaggerModule34::class,
      DaggerModule35::class, DaggerModule36::class, DaggerModule37::class, DaggerModule38::class,
      DaggerModule39::class, DaggerModule40::class, DaggerModule41::class, DaggerModule42::class,
      DaggerModule43::class, DaggerModule44::class, DaggerModule45::class, DaggerModule46::class,
      DaggerModule47::class, DaggerModule48::class, DaggerModule49::class, DaggerModule50::class,
      DaggerModule51::class, DaggerModule52::class, DaggerModule53::class, DaggerModule54::class,
      DaggerModule55::class, DaggerModule56::class, DaggerModule57::class, DaggerModule58::class,
      DaggerModule59::class, DaggerModule60::class, DaggerModule61::class, DaggerModule62::class,
      DaggerModule63::class, DaggerModule64::class, DaggerModule65::class, DaggerModule66::class,
      DaggerModule67::class, DaggerModule68::class, DaggerModule69::class, DaggerModule70::class,
      DaggerModule71::class, DaggerModule72::class, DaggerModule73::class, DaggerModule74::class,
      DaggerModule75::class, DaggerModule76::class, DaggerModule77::class, DaggerModule78::class,
      DaggerModule79::class, DaggerModule80::class, DaggerModule81::class, DaggerModule82::class,
      DaggerModule83::class, DaggerModule84::class, DaggerModule85::class, DaggerModule86::class,
      DaggerModule87::class, DaggerModule88::class, DaggerModule89::class]
)
public object ReplacingDaggerModule
```